src/main/java/com/xiao/users/service/impl/UserServiceImpl.java

```diff
@@ -23,17 +28,36 @@ public class UserServiceImpl implements IUserService {
 
     private final RoleMapper roleMapper;
 
-    public UserServiceImpl(UserRepository userRepository, UserMapper userMapper, RoleMapper roleMapper) {
-        this.userRepository = userRepository;
-        this.userMapper = userMapper;
-        this.roleMapper = roleMapper;
-    }
-
-    @Override
-    public void createUser(UserDto userDto) {
-        User user = userMapper.userDtoToUser(userDto);
-        userRepository.save(user);
-    }
+    private final AppConfig appConfig;
+
+    private final KafkaTemplate<String, String> kafkaTemplate;
+
+    private final ObjectMapper objectMapper;
+
+  public UserServiceImpl(
+      UserRepository userRepository,
+      UserMapper userMapper,
+      RoleMapper roleMapper,
+      AppConfig appConfig,
+      KafkaTemplate<String, String> kafkaTemplate,
+      ObjectMapper objectMapper) {
+    this.userRepository = userRepository;
+    this.userMapper = userMapper;
+    this.roleMapper = roleMapper;
+    this.appConfig = appConfig;
+    this.kafkaTemplate = kafkaTemplate;
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public void createUser(UserDto userDto) throws JsonProcessingException {
+    User user = userMapper.userDtoToUser(userDto);
+    User createdUser = userRepository.save(user);
+    kafkaTemplate.send(
```

Comment:
should extract kafka sending to a sperate service

Reply:
Created new KafkaService class